xlml/apis/gcp_config.py

```diff
@@ -32,7 +33,7 @@ class GCPConfig:
   """
 
   project_name: str
-  zone: str
+  zone: List[str]
```

Comment:
I'm kind of surprised that some checks didn't fail with this change, since it touches basically every single test in this repository, which have not been updated to use a list. Thanks [duck typing](https://en.wikipedia.org/wiki/Duck_typing).

Tying a retry policy to the `GCPConfig` doesn't seem right to me. It seems like it's mixing levels of abstraction in a way that doesn't seem healthy.

Retry policies that use different zones seem like a headache in general: how do you handle regionalized depedencies like networks or data? But, if we absolutely have to go down this road, I think it makes more sense to leave `GCPConfig` as-is and instead have your `Task` take a list of `GCPConfig`s if it supports retries. Retries may need to take place in multiple projects, or across multiple GKE clusters (which will soon be captured as an extension of GCPConfig).